packages/runtime/container-runtime/src/compatUtils.ts

```diff
@@ -0,0 +1,163 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { FlushMode } from "@fluidframework/runtime-definitions/internal";
+// The semver package documents and encourages these imports for users that only need some of the semver functionality.
+// eslint-disable-next-line import/no-internal-modules
+import semverGte from "semver/functions/gte.js";
+// eslint-disable-next-line import/no-internal-modules
+import semverLte from "semver/functions/lte.js";
+// eslint-disable-next-line import/no-internal-modules
+import semverValid from "semver/functions/valid.js";
+
+import {
+	disabledCompressionConfig,
+	enabledCompressionConfig,
+} from "./compressionDefinitions.js";
+import { type IContainerRuntimeOptionsInternal } from "./containerRuntime.js";
+import { pkgVersion } from "./packageVersion.js";
+
+/**
+ * Our policy is to support N/N-1 compatibility by default, where N is the most recent public major release of the runtime.
+ * Therefore, if the customer does not provide a compatibility mode, we will default to use N-1.
+ *
+ * However, this is not consistent with today's behavior. Some options (i.e. batching, compression) are enabled by default
+ * despite not being compatible with 1.x clients. Since the policy was introduced during 2.x's lifespan, N/N-1 compatibility
+ * by **default** will be in effect starting with 3.0. Importantly though, N/N-1 compatibility is still guaranteed with the
+ * proper configurations set.
+ *
+ */
+export const defaultCompatibilityMode = "2.0.0" as const;
```

Comment:
i've been dealing with some cross layer compat, and i think we'll have to rename this all to be more specific, probably something like protocolCompatailityVersion, as right now it is very confusing that compat mean for each of our different compat subsystem.

Reply:
I'm planning on renaming all the compatibilityMode properties in the next PR anyway which adds a public API surface. I think we should discuss that once that PR is open